### PR TITLE
Migration/accidental

### PR DESCRIPTION
**Pull request by rvilarl**
Accidental! @wassertim asked me to take it over.

@0xfe @ronyeh we are almost there Vex & Factory are waiting for the pendig merges :)

### CODE REVIEW

File: src/accidental.ts [hunk 1]
```diff
@@ -17,38 +17,75 @@ import { Modifier } from './modifier';
...
+export interface Line {
+  column: number;
+  line: number;
+  flatLine: boolean;
+  dblSharpLine: boolean;
+  numAcc: number;
+  width: number;
+}
+
+export interface AccidentalListItem {
```
Comment (0xfe) [inline src/accidental.ts:34]: This can just be an internal type alias. No need for interface.

File: src/accidental.ts [hunk 2]
```diff
@@ -17,38 +17,75 @@ import { Modifier } from './modifier';
 import { Glyph } from './glyph';
 import { GraceNoteGroup } from './gracenotegroup';
 import { GraceNote } from './gracenote';
+import { ModifierContextState } from './modifiercontext';
+import { Voice } from './voice';
+import { Note } from './note';
+import { StaveNote } from './stavenote';
+
+export interface Line {
```
Comment (0xfe) [inline src/accidental.ts:25]: This can also be an internal type alias.

File: src/accidental.ts [hunk 3]
```diff
@@ -526,17 +562,17 @@ export class Accidental extends Modifier {
     L('Rendering: ', type, accX, accY);
 
     if (!cautionary) {
-      glyph.render(context, accX, accY);
+      glyph?.render(ctx, accX, accY);
```
Comment (0xfe) [inline src/accidental.ts:565]: These `?` in this section are all potential silent failures. Can we explicitly check for them? (Or better still not make them optional?)

Comment (rvilarl) [reply src/accidental.ts:565]: Done.

File: src/accidental.ts (hunk 1 above)
Comment (rvilarl) [reply src/accidental.ts:34]: done

File: src/accidental.ts (hunk 2 above)
Comment (rvilarl) [reply src/accidental.ts:25]: done

**Comment by rvilarl** [comment]
Ready

**Comment by 0xfe** [comment]
Looks like there's conflicts from that last merge. :-)

**Comment by rvilarl** [comment]
Conflict resolved and Flow.accidentalColumnsTable type definition fixed.

**Comment by rvilarl** [comment]
@0xfe ready to merge

File: src/accidental.ts [hunk 4]
```diff
@@ -17,23 +17,61 @@ import { Modifier } from './modifier';
...
+type Line = {
+  column: number;
+  line: number;
+  flatLine: boolean;
+  dblSharpLine: boolean;
+  numAcc: number;
+  width: number;
+};
+
+type AccidentalListItem = {
```
Comment (0xfe) [inline src/accidental.ts:34]: Can you move these types into `format` where they are used? (type aliases are allowed inside function bodies)

**Comment by rvilarl** [comment]
Ready

File: src/accidental.ts (hunk 4 above)
Comment (rvilarl) [reply src/accidental.ts:34]: Line must stay out as it is used on other functions.

**Comment by rvilarl** [comment]
Now with Line global, hopefully ready

File: src/accidental.ts [hunk 5]
```diff
@@ -528,17 +565,17 @@ export class Accidental extends Modifier {
     L('Rendering: ', type, accX, accY);
 
     if (!cautionary) {
-      glyph.render(context, accX, accY);
+      glyph.render(ctx, accX, accY);
     } else {
       // Render the accidental in parentheses.
-      parenRight.render(context, accX, accY);
-      accX -= getGlyphWidth(parenRight);
+      parenRight?.render(ctx, accX, accY);
```
Comment (0xfe) [inline src/accidental.ts:571]: These are silent failures here. Can you explicitly test them please? (Maybe with `check`?)

(The ones below too.)

Comment (rvilarl) [reply src/accidental.ts:571]: Done

**Comment by rvilarl** [comment]
Ready hopefully :)

File: src/accidental.ts [hunk 6]
```diff
@@ -528,17 +565,17 @@ export class Accidental extends Modifier {
...
 
     if (!cautionary) {
-      glyph.render(context, accX, accY);
+      glyph.render(ctx, accX, accY);
     } else {
       // Render the accidental in parentheses.
-      parenRight.render(context, accX, accY);
-      accX -= getGlyphWidth(parenRight);
+      check<Glyph>(parenRight).render(ctx, accX, accY);
+      accX -= check<Glyph>(parenRight).getMetrics().width ?? 0;
```
Comment (0xfe) [inline src/accidental.ts:572]: the `?? 0` is a silent failure again -- the previous code expects it to be defined, no?

Can we remove the `?? 0` here and in the other `getGlyphWidth` replacements?

Comment (rvilarl) [reply src/accidental.ts]: This is no longer required, width is now not optional. Removed.

**Comment by rvilarl** [comment]
@0xfe ready to merge

**Comment by 0xfe** [comment]
Great!

**Pull request merged**